RFC-0039-generalize-pytorch-ut.md

```diff
@@ -0,0 +1,52 @@
+
+# [RFC] Generalization of PyTorch framework UT for non-cuda device execution
+
+**Authors:**
+* @ankurneog
+
+
+## **Summary**
+Modify PyTorch framework UTs so that non-cuda devices such as intel Gaudi and intel XPU is able to harness the content and improve quality.
+
+
+## **Motivation**
+The Pytorch framework UTs are good indicator for device stack health, however these are mostly written for cpu and cuda devices, which restricts its use for non-cuda devices.
+
+We propose to modify the content wherever possible to make it available for non-cuda device execution
+
+This will also ensure greater participation for content enhancement.
+
+## **Proposed Implementation**
+Since the content is huge, we propose a staggered approach for the implementation
+Steps:
+*   Remove restriction imposed through @onlyNativeDevices in core content, replace these with hooks so that supported devices can enable their content selectively.
+These should be flexible enough to support both in-tree and out-of-tree devices.
+*   Dtypes for a device should be dynamically loaded per op based on a common dictionary, instead of using different variables per device , eg: dtypesIfCuda
+* Miscelleneous decorators such as @skipIfCuda should be generalized @skipIfDevice
+*   Extend use of instantiate_device_type for all content, so that developers are forced to use generalized device code rather than using "cuda" or "cpu"
+* Generalize common distributed content , so that it can be extended for non nccl backends such as intel's hccl and ccl
```

Comment:
cc @wconstab if there is anyone on the distributed side that can help with designing this?